python/pyspark/worker.py

```diff
@@ -596,6 +599,10 @@ def process():
             profiler.profile(process)
         else:
             process()
+
+        # reset task context to None
+        TaskContext._setTaskContext(None)
+        BarrierTaskContext._setTaskContext(None)
```

Comment:
Hm, what happens if it fails with exceptions in the middle of execution?

Reply:
Is it really needed? We always set the global TaskContext and never reset it previouslly.

Reply:
> Hm, what happens if it fails with exceptions in the middle of execution in this worker?
If exceptions occured, the worker will be closed with `sys.exit(-1)`. 

> Is it really needed? We always set the global TaskContext and never reset it previouslly.
Previously:
```
val rdd = ...
val barriered = rdd.barrier().mapPartitions(...)

barriered.mapPartitions(...)  # here the BarrierTaskContext still existed.
```

This code is just a guard program, it shouldn't increase extra overhead or behavior change.

Reply:
> If exceptions occured, the worker will be closed with `sys.exit(-1)`.

I meant, this guard logic will be ignored anyway in that case. Why do we need this if we should only clean up when it succeeds?

Reply:
> This code is just a guard program, it shouldn't increase extra overhead or behavior change.

I guess that's only when the worker is reused. Can you clarify it with comments here?